Raven.Database/Actions/DocumentActions.cs

```diff
@@ -273,6 +273,8 @@ public int BulkInsert(BulkInsertOptions options, IEnumerable<IEnumerable<JsonDoc
                     cts.Token.ThrowIfCancellationRequested();
 
                     var docsToInsert = docs.ToArray();
+                    docsToInsert.ForEach(document => document.Key = string.IsNullOrWhiteSpace(document.Key) ? Guid.NewGuid().ToString() : document.Key.Trim());
```

Comment:
I think it will be better to move this code here somewhere around here
https://github.com/ravendb/ravendb/blob/v3.5/Raven.Database/Actions/DocumentActions.cs#L292
by doing this you will not iterate twice the docsToInsert

Also do we want to support Guid.NewGuid() ? This is not part of the issue and I do not see a test for this case. This line was throwing if the key was null and now it will be false always: https://github.com/ravendb/ravendb/blob/v3.5/Raven.Database/Actions/DocumentActions.cs#L293